### PR TITLE
Fix daml-doc `md`

### LINKED ISSUE #16474: Fix daml-doc md local files linking to html
Generating a folder with daml-docs and `--format=md` gives `md` files that link to eachother as `html` files, and as such, do not work.

### PR DESCRIPTION
**Pull request by samuel-williams-da**
Resolves #16474 
Relevant question: https://discuss.daml.com/t/markdown-referencing-html-in-generated-doc/6222

Simply switches the generated `md` files and hoogle db to use `.md` on internal global anchors, when run with `format=md`

### CODE REVIEW

**Comment by samuel-williams-da** [comment]
Md fixes not tested, waiting on information for how to run hoogle with the generated file

**Comment by samuel-williams-da** [comment]
Confirmed hoogle points to `md` file when `format=md` is used.

**Comment by samuel-williams-da** [comment]
Fair point, I can add a `--doc-ext` flag that defaults to html?

**Comment by samuel-williams-da** [comment]
Seems like odd UX though to generate `md` docs and have it pointing to `html`, perhaps its worth somehow enquiring on whether people do this?

**Comment by samuel-williams-da** [comment]
Alternatively, I may add a little warning note if you use format=md without doc-ext=md. Do you think thats reasonable?

**Comment by akrmn** [comment]
yeah honestly I'm not sure what's the best for UX here. At least we don't seem to be using `daml damlc docs` to produce markdown in the DA repos (see [search](https://github.com/search?q=org%3Adigital-asset+%22daml+damlc+docs%22&type=code)). The least disruptive would be to set `--doc-ext=html` by default, I think

**Comment by samuel-williams-da** [comment]
Yeah I believe our own docs are generated by calling the internals directly

**Comment by samuel-williams-da** [comment]
What do you think about the warning?

**Comment by akrmn** [comment]
Also, note that the paths with `.html` appear unused in the `rst` codepath, see [here](https://github.com/digital-asset/daml/blob/main/compiler/damlc/daml-doc/src/DA/Daml/Doc/Render/Rst.hs#L75)

**Comment by akrmn** [comment]
> What do you think about the warning?
> add a little warning note if you use format=md without doc-ext=md

hm yeah that sounds fine too

**Comment by samuel-williams-da** [comment]
Do we consider `format=rst` and `doc-ext` defined an error, given it will have no effect?

**Comment by akrmn** [comment]
> Do we consider `format=rst` and `doc-ext` defined an error, given it will have no effect?

I think it's fine if it's just ignored in that case

File: compiler/damlc/lib/DA/Cli/Damlc/Command/Damldoc.hs [hunk 1]
```diff
@@ -151,7 +153,7 @@ documentation numProcessors = Damldoc
     optOutputFormat =
         option readOutputFormat $
             metavar "FORMAT"
-            <> help "Output format. Valid format names: rst, md, markdown, html, hoogle, json (Default: markdown)."
+            <> help "Output format. Valid format names: rst, md, markdown, html, json (Default: markdown)."
```
Comment (akrmn) [inline compiler/damlc/lib/DA/Cli/Damlc/Command/Damldoc.hs:156]: thanks for spotting this :)

**Review by akrmn — APPROVED** [review]

**Pull request merged**